.ci/create_jenkins_job.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash +ex
+
+# Creates a new Jenkins job from a template, enables it and builds it.
+#
+# Uses Jenkins CLI (https://jenkins.io/doc/book/managing/cli/)
+# Expects $JENKINS_URL (url of Jenkins) and $JENKINS_CLI_JAR (path to jenkins-cli.jar) to be set as environment variables
+#
+# Usage: create_jenkins_job.sh <new_job_name> <template_job_name> [<view_name>]
+#
+# Example: create_jenkins_job.sh $BUILD_PARAMETER template demos
+
+# ANSI color codes
+RED="\e[31m"
+GREEN="\e[32m"
+BLUE="\e[34m"
+NC="\e[0m"
+
+NEW_JOB_NAME=$1
+TEMPLATE_JOB_NAME=$2
+VIEW_NAME=$3
+
+if [[ ! "$NEW_JOB_NAME" =~ ^[a-z0-9]([a-z0-9-]*[a-z0-9])?$ ]]; then
+    echo -e "${RED}Invalid project name!${NC}"
+    echo -e "${BLUE}Use only lowercase letters and numbers, you can include dashes in the middle (eg. \"my-demo-42\").${NC}"
+    exit 1
+fi
+if [[ -z "$JENKINS_URL" ]]; then
+    echo -e "${RED}Jenkins URL has not been provided (in \$JENKINS_URL environment variable)!${NC}"
+    exit 1
+fi
+if [[ -z "$JENKINS_CLI_JAR" ]]; then
+    echo -e "${RED}Path to the Jenkins CLI jarfile has not been provided (in \$JENKINS_CLI_JAR environment variable)!${NC}"
+    exit 1
+fi
+
+echo -e "${BLUE}Creating a new project \"$NEW_JOB_NAME\"...${NC}"
+
+java -jar "$JENKINS_CLI_JAR" -s "$JENKINS_URL" copy-job "$TEMPLATE_JOB_NAME" "$NEW_JOB_NAME"
+if [[ ! $? -eq 0 ]]; then
```

Comment:
Have you read the linked [Jenkins CLI docs](https://jenkins.io/doc/book/managing/cli/)?

`java -jar jenkins-cli.jar -s http://127.0.0.1:8080 copy-job job new-job` just copies a job into a new-job using the Jenkins CLI, the line above does that using variables instead of hard-coded parameters.

`if [[ ! $? -eq 0 ]]; then` is just a condition "if the last command's exit code is not 0" so basically "if the last command failed"